api/envoy/config/common/mutation_rules/v3/mutation_rules.proto

```diff
@@ -0,0 +1,77 @@
+syntax = "proto3";
+
+package envoy.config.common.mutation_rules.v3;
+
+import "envoy/type/matcher/v3/regex.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.config.common.mutation_rules.v3";
+option java_outer_classname = "MutationRulesProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/config/common/mutation_rules/v3;mutation_rulesv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// The HeaderMutationRules structure specifies what headers may be
+// manipulated by a processing filter. This set of rules rule makes it
+// possible to control which modifications a filter may make according
+// to rules here.
+//
+// By default, an external processing server may add, modify, or remove
+// any header except for an "Envoy internal" header (which is typically
+// denoted by an x-envoy prefix) or specific headers that may affect
+// further filter processing:
+//
+// * host
+// * :authority
+// * :scheme
+// * :method
```

Comment:
`:path`?

Reply:
In the current implementation, modifications to :path are allowed by default. I don't think that we should change that now, as modifying the path is pretty common and doesn't have the potential effects on the rest of processing as modifications of scheme and authority, I don't think. We had a bunch of discussion about this early on but I'm not sure where I would find that archived today.

Reply:
In that case, the `allow_routing` name is misleading. `:path` clearly is part of the routing headers. I would agree that it should be allowed by default. Suggest instead `allow_extra_routing` or something to that effect.

Reply:
That makes sense. I changed the name to allow_all_routing instead. I think that's an improvement.